google/api_core/path_template.py

```diff
@@ -298,6 +298,26 @@ def transcode(http_options, message=None, **request_kwargs):
         # Assign body and query params
         body = http_option.get("body")
 
+        # gapic-generator-python appends an underscore to field names
+        # that collide with python keywords. See related issue
+        # https://github.com/googleapis/python-api-core/issues/227
+        # Make sure that when the fields are REST encoded, the
+        # Python-specific trailing underscore is removed before going on the wire.
+        # `leftovers` can either be a dict or protobuf message.
+        # When `leftovers` is a dict, the `_` suffix in each key
+        # is stripped away as it is not possible to natively define a field
+        # with a trailing underscore in protobuf.
+        # When `leftovers` is a protobuf message, we need to use an underscore
+        # suffix when accessing the field in the protobuf message when the
+        # field has an underscore suffix.
+        field_suffix = ""
+
+        if isinstance(leftovers, dict):
+            leftovers = {key.rstrip("_"): val for key, val in leftovers.items()}
```

Comment:
Possible corner case here - are there instances where we should _not_ remove the trailing underscore from a field name? I'm not sure if a protobuf field name can end with an underscore and still be valid, but it might be worth adding some validation to ensure we're only stripping the suffix from the right fields.

Reply:
Based on comment https://github.com/googleapis/python-api-core/issues/227#issuecomment-880273391 it is not possible for fields to have trailing `_` at the proto level (Creating a field like this generates an error from the protobuf runtime).

Reply:
SGTM thanks for double-checking!